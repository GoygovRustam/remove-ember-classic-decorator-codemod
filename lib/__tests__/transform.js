const { transform } = require('../transform');

function generateSnapshot(source) {
  let result = transform(source);

  return `
========== INITIAL ============
${source}
=========== RESULT ============
${result}
_____________________________________________
`;
}

describe('Basic', () => {
  test('import statement and classic decorator', () => {
    let source = `
      import classic from 'ember-classic-decorator';
      import Component from '@ember/component';

      @classic
      export default class TestComponent extends Component {
      }
  `;

    expect(generateSnapshot(source)).toMatchSnapshot();
  });

  test('only init hook', () => {
    let source = `
      import Component from '@ember/component';

      export default class TestComponent extends Component {
          init() {
            super.init(...arguments)
          }
      }
  `;

    expect(generateSnapshot(source)).toMatchSnapshot();
  });

  test('import statement, classic decorator and init hook', () => {
    let source = `
      import classic from 'ember-classic-decorator';
      import Component from '@ember/component';

      @classic
      export default class TestComponent extends Component {
          init() {
            super.init(...arguments)
          }
      }
  `;

    expect(generateSnapshot(source)).toMatchSnapshot();
  });

  test('init hook with no arguments', () => {
    let source = `
      import Component from '@ember/component';

      export default class TestComponent extends Component {
          init() {
            super.init()
          }
      }
  `;

    expect(generateSnapshot(source)).toMatchSnapshot();
  });

  test('init hook with custom arguments', () => {
    let source = `
      import Component from '@ember/component';

      export default class TestComponent extends Component {
          init() {
            super.init(a, b, c)
          }
      }
  `;

    expect(generateSnapshot(source)).toMatchSnapshot();
  });

  test('already converted', () => {
    let source = `
      import Component from '@ember/component';

      export default class TestComponent extends Component {
          constructor() {
            super(a, b, c)
          }
      }
  `;

    expect(generateSnapshot(source)).toMatchSnapshot();
  });

  test('import statement, classic decorator and init hook when multiple decorators exist', () => {
    let source = `
      import classic from 'ember-classic-decorator';
      import Component from '@ember/component';

      @testDecorator
      @classic
      @anotherTestDecorator
      export default class TestComponent extends Component {
          init() {
            super.init(...arguments)
          }
      }
  `;

    expect(generateSnapshot(source)).toMatchSnapshot();
  });
});
