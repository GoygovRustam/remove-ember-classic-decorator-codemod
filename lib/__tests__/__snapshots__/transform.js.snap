// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic already converted 1`] = `
"
========== INITIAL ============

      import Component from '@ember/component';

      export default class TestComponent extends Component {
          constructor() {
            super(a, b, c)
          }
      }
  
=========== RESULT ============

      import Component from '@ember/component';

      export default class TestComponent extends Component {
          constructor() {
            super(a, b, c)
          }
      }
  
_____________________________________________
"
`;

exports[`Basic import statement and classic decorator 1`] = `
"
========== INITIAL ============

      import classic from 'ember-classic-decorator';
      import Component from '@ember/component';

      @classic
      export default class TestComponent extends Component {
      }
  
=========== RESULT ============

      import Component from '@ember/component';

      export default class TestComponent extends Component {
      }
  
_____________________________________________
"
`;

exports[`Basic import statement, classic decorator and init hook 1`] = `
"
========== INITIAL ============

      import classic from 'ember-classic-decorator';
      import Component from '@ember/component';

      @classic
      export default class TestComponent extends Component {
          init() {
            super.init(...arguments)
          }
      }
  
=========== RESULT ============

      import Component from '@ember/component';

      export default class TestComponent extends Component {
          constructor() {
            super(...arguments)
          }
      }
  
_____________________________________________
"
`;

exports[`Basic import statement, classic decorator and init hook when multiple decorators exist 1`] = `
"
========== INITIAL ============

      import classic from 'ember-classic-decorator';
      import Component from '@ember/component';

      @testDecorator
      @classic
      @anotherTestDecorator
      export default class TestComponent extends Component {
          init() {
            super.init(...arguments)
          }
      }
  
=========== RESULT ============

      import Component from '@ember/component';

      @testDecorator
      @anotherTestDecorator
      export default class TestComponent extends Component {
          constructor() {
            super(...arguments)
          }
      }
  
_____________________________________________
"
`;

exports[`Basic init hook with custom arguments 1`] = `
"
========== INITIAL ============

      import Component from '@ember/component';

      export default class TestComponent extends Component {
          init() {
            super.init(a, b, c)
          }
      }
  
=========== RESULT ============

      import Component from '@ember/component';

      export default class TestComponent extends Component {
          constructor() {
            super(a, b, c)
          }
      }
  
_____________________________________________
"
`;

exports[`Basic init hook with no arguments 1`] = `
"
========== INITIAL ============

      import Component from '@ember/component';

      export default class TestComponent extends Component {
          init() {
            super.init()
          }
      }
  
=========== RESULT ============

      import Component from '@ember/component';

      export default class TestComponent extends Component {
          constructor() {
            super()
          }
      }
  
_____________________________________________
"
`;

exports[`Basic only init hook 1`] = `
"
========== INITIAL ============

      import Component from '@ember/component';

      export default class TestComponent extends Component {
          init() {
            super.init(...arguments)
          }
      }
  
=========== RESULT ============

      import Component from '@ember/component';

      export default class TestComponent extends Component {
          constructor() {
            super(...arguments)
          }
      }
  
_____________________________________________
"
`;
